stan/math/prim/mat/meta/is_vector.hpp

```diff
@@ -2,27 +2,55 @@
 #define STAN_MATH_PRIM_MAT_META_IS_VECTOR_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/meta/is_eigen.hpp>
 #include <stan/math/prim/scal/meta/is_vector.hpp>
 
 namespace stan {
 
-// FIXME: use boost::type_traits::remove_all_extents to
-//        extend to array/ptr types
+namespace internal {
+// primary template for checking if eigen matrix is column vector
+template <class T, bool = is_eigen<std::decay_t<T>>::value>
+struct is_eigen_col_vector_impl
+    : std::integral_constant<bool, T::ColsAtCompileTime == 1> {};
 
+// not eigen type
+template <class T>
+struct is_eigen_col_vector_impl<T, false> : std::false_type {};
+}  // namespace internal
+
+// Check whether type is an eigen col vector
```

Comment:
We use an internal templated implementation here because if the preprocessor tries to access `ColsAtCompileTime` from a `T` of `double` it will throw. So this will only look for the columns at compile time member if the type is eigen